components/CategoricalField.vue

```diff
@@ -24,15 +24,15 @@ export default {
   props: {
     name: {
       type: String,
-      default: null,
+      required: true,
     },
     defaultSelected: {
       type: String,
       default: null,
     },
     options: {
       type: Array,
-      default: null,
+      default: Array,
```

Comment:
```suggestion
      default: [],
```
careful, you are referencing the builtin `Array` type constructor here, which will turn `options` into a pointer to the [`Array` constructor function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Array), and not initialize an empty array as you intended! 

There are a couple of ways to make [a new Array instance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#constructor) but the correct use here is going to be `default: []` because it is readable, short, and expected.

Reply:
Yea that's what I did when I first started implementing components and their props but according to [`vue/require-valid-default-prop`](https://eslint.vuejs.org/rules/require-valid-default-prop.html) the default for `Array` and `Object` must be returned using a function.
I had tried using the Array constructor itself but that triggers the `no-array-constructor` error from eslint so I'm passing a pointer to a function instead and the options were what you see here and
```
() => []
```

Reply:
Ah I see, my bad. I had forgotten about the Vue props validation. You are right that for Array types the default value cannot be a simple instance. What's wrong with `() => []`? That looks correct to me.

Reply:
Nothing just another way of doing the same thing

Reply:
Are you referring to this?
```
    propE: {
      type: Object,
      // Object or array defaults must be returned from
      // a factory function. The function receives the raw
      // props received by the component as the argument.
      default(rawProps) {
        return { message: 'hello' }
      }
    },
```

Reply:
yeah, although I think I shared the vue3 docs. Here is the relevant section for us: https://v2.vuejs.org/v2/guide/components-props.html#Prop-Validation

Reply:
Yea thanks for that I didn't know they mentioned that in their docs.

Reply:
NP, neither did I :) or rather forgot again. We do use this pattern for defaults in the annotation_tool.